bbot/modules/internal/excavate.py

```diff
@@ -303,21 +314,35 @@ class excavateTestRule(ExcavateRule):
     }
 
     options = {
-        "retain_querystring": False,
+        "retain_querystring": True,
         "yara_max_match_data": 2000,
         "custom_yara_rules": "",
+        "speculate_params": False,
     }
     options_desc = {
         "retain_querystring": "Keep the querystring intact on emitted WEB_PARAMETERS",
         "yara_max_match_data": "Sets the maximum amount of text that can extracted from a YARA regex",
         "custom_yara_rules": "Include custom Yara rules",
+        "speculate_params": "Enable speculative parameter extraction from JSON and XML content",
     }
     scope_distance_modifier = None
     accept_dupes = False
 
     _module_threads = 8
 
-    parameter_blacklist = {
+    parameter_blacklist_prefix = [
+        "TS01",
+        "BIGipServer",
+        "incap_",
+        "visid_incap_",
+        "AWSALB",
+        "utm_",
+        "ApplicationGatewayAffinity",
+        "JSESSIONID",
+        "ARRAffinity",
+    ]
+
+    parameter_blacklist = set(
```

Comment:
Can we move this param blacklist to the config, similar to how we did with the url blacklist? Allowing people to tweak it in case for whatever reason they want to fuzz one of these params.

Reply:
fixed in e9a9faaf77780999d91932e55f2b867239345852